### PR TITLE
Optimize for the leaf groups that has no children since there are mos…

### PR DESCRIPTION
**Pull request by baldersheim**
…t of them.

@jonmv or @bratseth or @bjorncs PR

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/search/grouping/vespa/ResultBuilder.java [hunk 1]
```diff
@@ -173,18 +173,23 @@ Group fill(Group group) {
...
+                    if (child != null) {
+                        group.add(child.build());
+                    }
                 }
             }
             return group;
         }
 
         GroupListBuilder getOrCreateChildList(int tag, boolean ranked) {
             int index = tag + 1; // Add 1 to avoid the dreaded -1 default value.
```
Comment (jonmv) [inline container-search/src/main/java/com/yahoo/search/grouping/vespa/ResultBuilder.java:187]: (((＼（✘෴✘）／)))

File: container-search/src/main/java/com/yahoo/search/grouping/vespa/ResultBuilder.java [hunk 2]
```diff
@@ -173,18 +173,23 @@ Group fill(Group group) {
...
-            if (index >= childLists.length) {
-                childLists = Arrays.copyOf(childLists, tag + 8);
+            if (childLists == null || index >= childLists.length) {
+                int reservedSize = (((index + 1) + (CHILDLIST_SIZE_INCREMENTS -1))/CHILDLIST_SIZE_INCREMENTS) * CHILDLIST_SIZE_INCREMENTS;
```
Comment (jonmv) [inline container-search/src/main/java/com/yahoo/search/grouping/vespa/ResultBuilder.java:189]:
```suggestion
                int reservedSize = ((index + CHILDLIST_SIZE_INCREMENTS) / CHILDLIST_SIZE_INCREMENTS) * CHILDLIST_SIZE_INCREMENTS;
```

**Review by jonmv — APPROVED** [review]
Consider the simplification of that one expression, otherwise 👍

**Review by bratseth — APPROVED** [review]

File: container-search/src/main/java/com/yahoo/search/grouping/vespa/ResultBuilder.java (hunk 2 above)
Comment (baldersheim) [reply container-search/src/main/java/com/yahoo/search/grouping/vespa/ResultBuilder.java]: I added minSize = index + 1 to make it clearer. I wanted to keep the (CHILDLIST_SIZE_INCREMENTS - 1) as together with the division and the multiplication is an often used pattern for rounding of integers.

**Pull request merged**